File: bin/cmds/scp.pl
#!/usr/bin/env perl

use strict;
use warnings;

my $port=shift(@ARGV);
my $password=shift(@ARGV);
my $args=join(" ",@ARGV);

open(P,"scp.exp $port $password $args |") or die;
my $l=0;
my $pw=0;
my $exitcode=0;
my $s="";
while (<P>) {
	++$l;
	if (($l==1) && /^spawn/) {
		$s.=$_;
		next;
	}
	if ($pw==0 && /assword:/) {
		$pw=1;
		$s.=$_;
		next;
	}
	if ($pw==0) {
		$s.=$_;
		next;
	}

	print STDERR $s;
	$s="";
	print STDERR $_;
	$exitcode=1
}
close(P);
if ($pw==0) {
	print STDERR $s;
	$exitcode=2;
}
exit $exitcode

File: bin/cmds/scp.py
#!/usr/bin/env python

import os
import sys
import re

def main():

    args = " "
    for s in sys.argv[1:]:
        args += ('\'' + s + '\' ')

    pw = False
    l = 0
    s = ""
    exitcode = 0

    r = os.popen("scp.exp" + args)
    lines = r.readlines()
    print lines
    for line in lines:
        l += 1
        line = line.replace('\r\n', '\n')
        if l == 1 and re.search('^spawn', line):
            s += line
            continue

        if pw is False and re.search('assword:', line):
            pw = True
            s += line
            continue

        if pw is False:
            s += line
            continue

        s += line
        exitcode = 1

    r.close()

    if pw is False:
        exitcode = 2

    if exitcode:
        sys.stderr.write(s)
        sys.stderr.write('exit ' + str(exitcode) + '\n')

    sys.exit(exitcode)

if __name__ == '__main__':
    main()